flash/vision/style_transfer/model.py

```diff
@@ -0,0 +1,227 @@
+        if self.upsample:
+            input = self.upsample(input)
+
+        output = self.conv(self.pad(input))
+
+        if self.norm:
+            output = self.norm(output)
+        if self.activation:
+            output = self.activation(output)
+
+        return output
+
+
+class Residual(nn.Module):
+
+    def __init__(self, channels: int) -> None:
+        super().__init__()
+        self.conv1 = Conv(channels, channels, kernel_size=3)
+        self.conv2 = Conv(channels, channels, kernel_size=3, activation=False)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        output = self.conv2(self.conv1(input))
+        return output + input
+
+
+class FloatToUint8Range(nn.Module):
+
+    def forward(self, input):
+        return input * 255.0
+
+
+class Uint8ToFloatRange(nn.Module):
+
+    def forward(self, input):
+        return input / 255.0
+
+
+class Transformer(nn.Module):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.encoder = nn.Sequential(
+            Conv(3, 32, kernel_size=9),
+            Conv(32, 64, kernel_size=3, stride=2),
+            Conv(64, 128, kernel_size=3, stride=2),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+        )
+        self.decoder = nn.Sequential(
+            Conv(128, 64, kernel_size=3, stride=2, upsample=True),
+            Conv(64, 32, kernel_size=3, stride=2, upsample=True),
+            Conv(32, 3, kernel_size=9, norm=False, activation=False),
+        )
+
+        self.preprocessor = FloatToUint8Range()
+        self.postprocessor = Uint8ToFloatRange()
+
+    def forward(self, input):
+        input = self.preprocessor(input)
+        output = self.decoder(self.encoder(input))
+        return self.postprocessor(output)
+
+
+class StyleTransfer(Task):
+
+    def __init__(
+        self,
+        style_image: Union[str, torch.Tensor],
+        model: Optional[nn.Module] = None,
+        multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None,
+        content_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        style_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        optimizer: Union[Type[torch.optim.Optimizer], torch.optim.Optimizer] = torch.optim.Adam,
+        optimizer_kwargs: Optional[Dict[str, Any]] = None,
+        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
+        scheduler_kwargs: Optional[Dict[str, Any]] = None,
+        learning_rate: float = 1e-3,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+    ):
+        self.save_hyperparameters(ignore="style_image")
+
+        if isinstance(style_image, str):
+            style_image = read_image(style_image)
+
+        if model is None:
+            # TODO: import this from pystiche
+            model = Transformer()
```

Comment:
`pystiche` should a hub of model (pretrained possibly) and make them available through the registry.

Reply:
Yes, it should, but it probably won't in the next time. I only have limited time / resources to maintain `pystiche` so it is only one model for now.